packages/ingest/src/granule.ts

```diff
@@ -458,6 +460,26 @@ export function unversionFilename(filename: string): string {
     : filename;
 }
 
+/**
+ * Generates a unique granule ID by appending a truncated MD5 hash of the granule object.
+ *
+ * @param granule - The granule object containing granuleId and collectionId.
+ * @param hashLength - The length of the hash to append to the granuleId.
+ * @returns - A unique granule ID in the format: granuleId_hash.
+ */
+export function generateUniqueGranuleId(granule: ApiGranule, hashLength: number): string {
+  // use MD5 to generate truncated hash of granule object
+  // in the format
+
+  const jsonString = JSON.stringify({
+    collectionId: granule.collectionId,
+    granuleId: granule.granuleId,
+    timestamp: process.hrtime.bigint().toString(),
+  });
+  const hashBuffer = crypto.createHash('md5').update(jsonString).digest();
+  return `${granule.granuleId}_${hashBuffer.toString('base64url').replace(/_/g, '').slice(0, hashLength)}`;
+}
+
```

Comment:
Do we also want a method that decodes the granuleId for granules that have already gone through this uniquification? Is that scoped elsewhere?

Reply:
The intention here is not to decode the value at all.

Reply:
We can talk about this in the architecture meeting. I'm confused by a line in the doc:
>Human-interpretable (users can infer the producerId *from* the granuleId)

Reply:
@npauzenga  it sounds like this was resolved in the arch meeting, let me know if that's not the case and I can respond.

Reply:
Yeah, for the GH record I was concerned that we couldn't extract the granuleId from the hash easily. And... yes, the granuleId is in the hash... but it's also in plaintext in the string before the hashed value 😄